primitives/session-manager/src/pallet_session_support.rs

```diff
@@ -0,0 +1,108 @@
+use log::debug;
+use sp_staking::SessionIndex;
+use sp_std::vec::Vec;
+
+/// Implements [`pallet_session::SessionManager`] and [`pallet_session::ShouldEndSession`] integrated with [`pallet_session_validator_management`].
+///
+/// To use it, wire it in runtime configuration of [`pallet_session`].
+#[allow(dead_code)]
+#[derive(new)]
+pub struct PalletSessionSupport<T> {
+	_phantom: PhantomData<T>,
+}
+
+impl<T: pallet_session_validator_management::Config + pallet_session::Config>
+	pallet_session::SessionManager<T::AccountId> for PalletSessionSupport<T>
+{
+	/// Sets the first validator-set by mapping the current committee from [`pallet_session_validator_management`]
+	fn new_session_genesis(_new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+		Some(
+			pallet_session_validator_management::Pallet::<T>::current_committee_storage()
+				.committee
+				.into_iter()
+				.map(|(id, _)| id.into())
+				.collect::<Vec<_>>(),
+		)
+	}
+
+	/// Rotates the committee in [`pallet_session_validator_management`] and plans this new committee as upcoming validator-set.
+	/// Updates the session index of [`pallet_session`].
+	// Instead of Some((*).expect) we could just use (*). However, we rather panic in presence of important programming errors.
+	fn new_session(new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+		debug!("PalletSessionSupport: New session {new_index}");
+		pallet_session::pallet::CurrentIndex::<T>::put(new_index);
+		Some(
+			pallet_session_validator_management::Pallet::<T>::rotate_committee_to_next_epoch()
+				.expect(
+					"PalletSessionSupport: Session should never end without current epoch validators defined. \
+				Check ShouldEndSession implementation or if it is used before starting new session",
+				).into_iter().map(|(id, _)| id.into()).collect::<Vec<_>>(),
+		)
+	}
+
+	fn end_session(end_index: SessionIndex) {
+		debug!("PalletSessionSupport: End session {end_index}");
+	}
+
+	// Session is expected to be at least 1 block behind sidechain epoch.
+	fn start_session(start_index: SessionIndex) {
+		let epoch_number = T::current_epoch_number();
+		debug!("PalletSessionSupport: Start session {start_index}, epoch {epoch_number}");
+	}
+}
+
+/// Tries to end each session in the first block of each partner chains epoch in which the committee for the epoch is defined.
+impl<T, EpochNumber> pallet_session::ShouldEndSession<BlockNumberFor<T>> for PalletSessionSupport<T>
+where
+	T: pallet_session_validator_management::Config<ScEpochNumber = EpochNumber>,
+	EpochNumber: Clone + PartialOrd,
+{
+	fn should_end_session(n: BlockNumberFor<T>) -> bool {
+		let current_epoch_number = T::current_epoch_number();
+
+		let should_end = current_epoch_number
+			> pallet_session_validator_management::Pallet::<T>::current_committee_storage().epoch
+			&& pallet_session_validator_management::Pallet::<T>::next_committee().is_some();
+		debug!("PalletSessionValidatorManagementSessionManager: should_end_session({n:?}) = {should_end}");
```

Comment:
Maybe we should log a warning if we can't switch epochs because of a missing committee?